homeassistant/components/fritz/binary_sensor.py

```diff
@@ -17,79 +20,76 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+SENSOR_TYPES = (
+    BinarySensorEntityDescription(
+        key="is_connected",
+        name="Connection",
+        device_class=DEVICE_CLASS_CONNECTIVITY,
+    ),
+    BinarySensorEntityDescription(
+        key="is_linked",
+        name="Link",
+        device_class=DEVICE_CLASS_PLUG,
+    ),
+    BinarySensorEntityDescription(
+        key="firmware_update",
+        name="Firmware Update",
+        device_class=DEVICE_CLASS_UPDATE,
+    ),
+)
+
+
 async def async_setup_entry(
     hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
 ) -> None:
     """Set up entry."""
     _LOGGER.debug("Setting up FRITZ!Box binary sensors")
     fritzbox_tools: FritzBoxTools = hass.data[DOMAIN][entry.entry_id]
 
-    if fritzbox_tools.connection and "WANIPConn1" in fritzbox_tools.connection.services:
+    if (
+        not fritzbox_tools.connection
+        or "WANIPConn1" not in fritzbox_tools.connection.services
+    ):
         # Only routers are supported at the moment
-        async_add_entities(
-            [FritzBoxConnectivitySensor(fritzbox_tools, entry.title)], True
-        )
+        return
+
+    entities = []
+    for sensor_type in SENSOR_TYPES:
+        entities.append(FritzBoxBinarySensor(fritzbox_tools, entry.title, sensor_type))
+
+    if entities:
+        async_add_entities(entities, True)
 
 
-class FritzBoxConnectivitySensor(FritzBoxBaseEntity, BinarySensorEntity):
+class FritzBoxBinarySensor(FritzBoxBaseEntity, BinarySensorEntity):
     """Define FRITZ!Box connectivity class."""
 
     def __init__(
-        self, fritzbox_tools: FritzBoxTools, device_friendly_name: str
+        self,
+        fritzbox_tools: FritzBoxTools,
+        device_friendly_name: str,
+        sensor_type: BinarySensorEntityDescription,
     ) -> None:
         """Init FRITZ!Box connectivity class."""
-        self._unique_id = f"{fritzbox_tools.unique_id}-connectivity"
-        self._name = f"{device_friendly_name} Connectivity"
-        self._is_on = True
-        self._is_available = True
+        self.entity_description: BinarySensorEntityDescription = sensor_type
```

Comment:
```suggestion
        self.entity_description = description
```
Type annotation is not necessary. Inherited from `BinarySensorEntity`.